equinix/resource_metal_device_acc_test.go

```diff
@@ -1134,21 +1134,21 @@ func testAccWaitForMetalDeviceActive(project, deviceHostName string) resource.Im
 
 		meta := testAccProvider.Meta()
 		rd := new(schema.ResourceData)
-		meta.(*config.Config).AddModuleToMetalUserAgent(rd)
-		client := meta.(*config.Config).Metal
-		devices, _, err := client.Devices.List(rs.Primary.ID, &packngo.ListOptions{Search: deviceHostName})
+		meta.(*config.Config).AddModuleToMetalGoUserAgent(rd)
+		client := meta.(*config.Config).Metalgo
+		devices, _, err := client.DevicesApi.FindProjectDevices(context.TODO(), rs.Primary.ID).Search(deviceHostName).Execute()
 		if err != nil {
 			return "", fmt.Errorf("error while fetching devices for project [%s], error: %w", rs.Primary.ID, err)
 		}
-		if len(devices) == 0 {
+		if len(devices.Devices) == 0 {
```

Comment:
To save a few lines of change, could have done something like:
```
resp := ...
devices := resp.GetDevices()
```

Reply:
Updated as described; resolving this thread but please reopen if needed.